x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsCanMatchOnCoordinatorIntegTests.java

```diff
@@ -0,0 +1,288 @@
+                )
+            );
+
+        if (includeIndexCoveringSearchRangeInSearchRequest) {
+            SearchResponse searchResponse = client().search(request).actionGet();
+
+            // All the regular index searches succeeded
+            assertThat(searchResponse.getSuccessfulShards(), equalTo(indexWithinSearchRangeShardCount));
+            // All the searchable snapshots shard search failed
+            assertThat(searchResponse.getFailedShards(), equalTo(indexOutsideSearchRangeShardCount));
+            assertThat(searchResponse.getSkippedShards(), equalTo(0));
+            assertThat(searchResponse.getTotalShards(), equalTo(totalShards));
+        } else {
+            // All shards failed, since all shards are unassigned and the IndexMetadata min/max timestamp
+            // is not available yet
+            expectThrows(SearchPhaseExecutionException.class, () -> client().search(request).actionGet());
+        }
+
+        // Allow the searchable snapshots to be finally mounted
+        unblockNode(repositoryName, dataNodeHoldingSearchableSnapshot);
+        waitUntilRecoveryIsDone(searchableSnapshotIndexOutsideSearchRange);
+        ensureGreen(searchableSnapshotIndexOutsideSearchRange);
+
+        final IndexMetadata updatedIndexMetadata = getIndexMetadata(searchableSnapshotIndexOutsideSearchRange);
+        final IndexLongFieldRange updatedTimestampMillisRange = updatedIndexMetadata.getTimestampMillisRange();
+        assertThat(updatedTimestampMillisRange.isComplete(), equalTo(true));
+        assertThat(updatedTimestampMillisRange, not(sameInstance(IndexLongFieldRange.EMPTY)));
+        assertThat(updatedTimestampMillisRange.getMin(), greaterThanOrEqualTo(Instant.parse("2020-11-26T00:00:00Z").getMillis()));
+        assertThat(updatedTimestampMillisRange.getMax(), lessThanOrEqualTo(Instant.parse("2020-11-27T00:00:00Z").getMillis()));
+        assertThat(indicesService.getTimestampFieldType(updatedIndexMetadata.getIndex()), notNullValue());
+
+        // Stop the node holding the searchable snapshots, and since we defined
+        // the index allocation criteria to require the searchable snapshot
+        // index to be allocated in that node, the shards should remain unassigned
+        internalCluster().stopNode(dataNodeHoldingSearchableSnapshot);
+        waitUntilAllShardsAreUnassigned(updatedIndexMetadata.getIndex());
+
+        if (includeIndexCoveringSearchRangeInSearchRequest) {
+            SearchResponse newSearchResponse = client().search(request).actionGet();
+
+            assertThat(newSearchResponse.getSkippedShards(), equalTo(indexOutsideSearchRangeShardCount));
+            assertThat(newSearchResponse.getSuccessfulShards(), equalTo(totalShards));
+            assertThat(newSearchResponse.getFailedShards(), equalTo(0));
+            assertThat(newSearchResponse.getTotalShards(), equalTo(totalShards));
+        } else {
+            if (indexOutsideSearchRangeShardCount == 1) {
```

Comment:
I am not sure I get this bit. I would have expected this to simply return a response similar to the case where we include the index covering search range in search request, except the totalShards would be `indexOutsideSearchRangeShardCount`? It is simply a no hit, since we search a set of indices where no data matches the timestamp criteria.

Now reading some of the code above I understand this :slightly_smiling_face:, let us tackle improving this in a follow-up.

Reply:
This is the current search behaviour, we might want to improve that and somehow provide the empty response directly from the coordinator, I'm not sure how involved is that, maybe @jimczi can guide us here?